sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -259,6 +261,17 @@ public static BlobItem populateBlobItem(BlobItemInternal blobItemInternal) {
         return blobItem;
     }
 
+    public static TaggedBlobItem populateTaggedBlobItem(FilterBlobItem filterBlobItem) {
+        Map<String, String> tags = new HashMap<>();
```

Comment:
Thoughts on deferring instantiation of a new map until we know the `FilterBlobItem` has tags, if it doesn't use the singleton `Collections.emptyMap()`.

Reply:
This translates responses of an API where you filter blobs by tags. Unless you're purposefully using an older service version, tags should never be empty.

Reply:
Given a refactor Gauri suggested, it may actually be good to do this, as the guarantee I presented would no longer be applicable.

Reply:
yeah, we shouldn't create empty maps if not required.